File: DSA_Java/2784.java
import java.util.HashMap;

class Solution {
    public boolean isGood(int[] nums) {
        HashMap<Integer,Integer> count=new HashMap<>();
        int n=nums.length;
        int max_num=nums[0];

        for(int i:nums){
            max_num=Math.max(i,max_num);
        }
        for(int i:nums){
            count.put(i,count.getOrDefault(i,0)+1);
        }
        for(int i=1;i<max_num+1;i++){
            int freq=count.getOrDefault(i,0);
            if (i==max_num){
                if (freq!=2){
                    return false;
                }
            }
            else{
                if (freq!=1){
                    return false;
                }
                
            }
        }
        return true;
        
    }

}

File: 2784.py
class Solution:
    def isGood(self, nums: List[int]) -> bool:
        count={}
        n=len(nums)
        max_n=max(nums)
        for num in nums:
            count[num] = count.get(num, 0) + 1
        for i in range(1,max_n+1):
            freq=count.get(i,0)
            if i==max_n:
                if freq!=2:
                    return False
            else:
                if freq!=1:
                    return False
        return True
 
                
        
        